services/StatsService.js

```diff
@@ -0,0 +1,144 @@
+        (SELECT COUNT(*) FROM "Comments") AS totalComments,
+        (SELECT COUNT(*) FROM "Projects") AS totalProjects,
+        (SELECT AVG("endDate"::date - "startDate"::date) FROM "Projects") AS averageLifespan, 
+        (
+          SELECT t."name"
+          FROM "ProjectTypes" pt
+          JOIN "Types" t ON t."id" = pt."typeId"
+          GROUP BY pt."typeId", t."name"
+          ORDER BY COUNT(*) DESC
+          LIMIT 1
+        ) AS mostCommonCauses,
+        (SELECT COUNT(*) FROM "Projects" WHERE "createdAt"::date = CURRENT_DATE) AS funeralsToday,
+        (SELECT COUNT(*) FROM "Upvotes") AS totalVotes,
+        (SELECT COUNT(*) FROM "Upvotes" WHERE "createdAt"::date = CURRENT_DATE) AS votesToday,
+        (SELECT AVG(LENGTH("eulogy")) FROM "Projects") AS averageEulogyLength,
+        (
+          SELECT COUNT(*) 
+          FROM "Projects" 
+          WHERE DATE_PART('day', "endDate" - "startDate") <= 1
+        ) * 100.0 / NULLIF((SELECT COUNT(*) FROM "Projects"), 0) AS rageQuitRate,
+        (
+          SELECT TRIM(TO_CHAR("endDate", 'Day')) AS topBurialDay
+          FROM "Projects"
+          WHERE "endDate" IS NOT NULL
+          GROUP BY topBurialDay
+          ORDER BY COUNT(*) DESC
+          LIMIT 1
+        ) AS topBurialDay,
+                 (
+          SELECT TRIM(TO_CHAR("endDate", 'Month')) AS topBurialMonth
+          FROM "Projects"
+          WHERE "endDate" IS NOT NULL
+          GROUP BY topBurialMonth
+          ORDER BY COUNT(*) DESC
+          LIMIT 1
+        ) AS topBurialMonth,
+        (
+        SELECT jsonb_build_object(
+          'id', p."id",
+          'name', p."name",
+          'votes', COUNT(u."id")
+        )
+        FROM "Projects" p
+        LEFT JOIN "Upvotes" u ON p."id" = u."projectId"
+        GROUP BY p."id", p."name"
+        ORDER BY COUNT(u."id") DESC
+        LIMIT 1
+      ) AS topVotedProject
+    `,
+      {
+        type: this.client.QueryTypes.SELECT,
+      }
+    );
+
+    return stats[0];
+  }
+
+  async getUserStats(userId) {
+    if (!userId) {
+      throw new Error("User ID is required");
+    }
+
+    const stats = await this.client.query(
+      `
+    SELECT
+      (SELECT COUNT(*) FROM "Projects" WHERE "userId" = :userId) AS totalProjects,
+      (SELECT COUNT(*) FROM "Comments" WHERE "userId" = :userId) AS totalComments,
+      (SELECT COUNT(*) FROM "Upvotes" WHERE "userId" = :userId) AS totalVotes,
+      (SELECT AVG(LENGTH("eulogy")) FROM "Projects" WHERE "userId" = :userId) AS averageEulogyLength,
+      (SELECT AVG("endDate"::date - "startDate"::date) FROM "Projects" WHERE "userId" = :userId) AS averageLifespan,
+      (
+        SELECT t."name"
+        FROM "ProjectTypes" pt
+        JOIN "Types" t ON t."id" = pt."typeId"
+        JOIN "Projects" p ON p."id" = pt."projectId"
+        WHERE p."userId" = :userId
+        GROUP BY pt."typeId", t."name"
+        ORDER BY COUNT(*) DESC
+        LIMIT 1
+      ) AS mostCommonCause,
```

Comment:
getUserStats in controller uses plural:

```
     mostCommonCauses: data.mostcommoncauses
```

Wonder if it needs to be the same in both? If so either change controller end elsewhere to use singular or this to use plural:  
```suggestion
      ) AS mostCommonCauses,
```

Reply:
I've fixed the miss-match on that one and the miss-match in the controllers, the get user/me stats outputs uses singular in the query but were using plural in the object creation output in the response so they weren't being added.